gdal/ogr/ogrsf_frmts/flatgeobuf/ogrflatgeobuflayer.cpp

```diff
@@ -398,68 +398,74 @@ OGRFeature *OGRFlatGeobufLayer::GetFeature(GIntBig nFeatureId)
+            OGREnvelope env;
+            m_poFilterGeom->getEnvelope(&env);
+            Rect r { env.MinX, env.MinY, env.MaxX, env.MaxY };
+            CPLDebug("FlatGeobuf", "Spatial index search on %f,%f,%f,%f", env.MinX, env.MinY, env.MaxX, env.MaxY);
+            auto readNode = [this, headerSize] (uint8_t *buf, size_t i, size_t s) {
+                if (VSIFSeekL(m_poFp, sizeof(magicbytes) + sizeof(uoffset_t) + headerSize + i, SEEK_SET) == -1)
+                    throw std::runtime_error("I/O seek failure");
+                if (VSIFReadL(buf, 1, s, m_poFp) != s)
+                    throw std::runtime_error("I/O read file");
+            };
+            m_foundFeatureIndices = PackedRTree::streamSearch(featuresCount, indexNodeSize, r, readNode);
+            m_featuresCount = m_foundFeatureIndices.size();
+            CPLDebug("FlatGeobuf", "%lu features found in spatial index search", static_cast<long unsigned int>(m_featuresCount));
+            m_queriedSpatialIndex = true;
+        }
+
+        if (!m_featureOffsets) {
+            CPLDebug("FlatGeobuf", "Seek to feature offsets index position");
+            if (VSIFSeekL(m_poFp, sizeof(magicbytes) + sizeof(uoffset_t) + headerSize + treeSize, SEEK_SET) == -1)
+                return CPLErrorIO();
+            m_featureOffsets = static_cast<uint64_t *>(VSI_MALLOC_VERBOSE(featureOffsetsSize));
+            if (!m_featureOffsets)
+                return CPLErrorMemoryAllocation();
+            CPLDebug("FlatGeobuf", "Reading feature offsets index");
+            if (VSIFReadL(m_featureOffsets, 8, featureOffetsCount, m_poFp) != featureOffetsCount)
+                return CPLErrorIO();
         }
-        CPLDebug("FlatGeobuf", "%lu features found in spatial index search", static_cast<long unsigned int>(m_featuresCount));
-        if (VSIFSeekL(m_poFp, sizeof(magicbytes) + sizeof(uoffset_t) + headerSize + treeSize, SEEK_SET) == -1)
-            return CPLErrorIO();
     } catch (const std::exception &e) {
-        CPLError(CE_Failure, CPLE_AppDefined, "querySpatialIndex: Unexpected failure: %s", e.what());
+        CPLError(CE_Failure, CPLE_AppDefined, "readIndex: Unexpected failure: %s", e.what());
         return OGRERR_FAILURE;
     }
-    if (m_featureOffsets) {
-        VSIFree(m_featureOffsets);
-        m_featureOffsets = nullptr;
-    }
-    auto featureOffetsCount = static_cast<size_t>(featuresCount);
-    auto featureOffsetsSize = featureOffetsCount * 8;
-    m_featureOffsets = static_cast<uint64_t *>(VSI_MALLOC_VERBOSE(featureOffsetsSize));
-    if (!m_featureOffsets)
-        return CPLErrorMemoryAllocation();
-    if (VSIFReadL(m_featureOffsets, 8, featureOffetsCount, m_poFp) != featureOffetsCount)
-        return CPLErrorIO();
-    m_queriedSpatialIndex = true;
+
     return OGRERR_NONE;
 }
 
-GIntBig OGRFlatGeobufLayer::GetFeatureCount(int bForce) {
+GIntBig OGRFlatGeobufLayer::GetFeatureCount(int bForce)
+{
```

Comment:
Then, all those `if ... {` may be worth to keep consistent too

Reply:
I follow the principle of function signatures being the exception to the rule to keep braces on the same line, same as in linux kernel I believe.